### PR TITLE
NGSIv1 removal

### PR DESCRIPTION
**Pull request by fgalan**
~This PR is not for being merged, it is just to expose publicly the branch. PR will be closed (and the branch deleted) after CI e2e testing.~ At the end code modification and test were added.

### CODE REVIEW

**Comment by jason-fox** [comment]
@fgalan Once #561 , #558 and #553 have been merged you may want to pull the changes into this branch to see if the NGSI-v1 deletion is working as expected.

**Comment by fgalan** [comment]
> @fgalan Once #561 , #558 and #553 have been merged you may want to pull the changes into this branch to see if the NGSI-v1 deletion is working as expected.

After merging all these PRs and upgrade this PR branch with master, tests pass to ok. Thus, as far as I understand the NGSIv1 removal work in IOTA-JSON code has ended.

(Let's have this PR opened while the overall NGSIv1 removal work is completed)

**Comment by fgalan** [comment]
Not sure if the NGSI-v1 removal work has ended in this repo... we have recently discovered isCurrentNgsi() usages in IOTA code. We are investigating it.

**Comment by jason-fox** [comment]
The `isCurrentNgsi()` method hasn't been removed, but it will always be `true`, so the NGSI-v1 specific stuff in the JSON IoT Agent will be uncovered lava code. It is up to you which way round you want to break the code base:

- iot-agent-node-lib doesn't support NGSI-v1 but the IoT Agent does.
-  IoT Agent doesn't support NGSI-v1 but iot-agent-node-lib  does.

Obviously such a situation is only temporary.

**Comment by jason-fox** [comment]
I have created (yet another) branch which removes the `isCurrent()` method - you could refer to it in the `package.json` to see where the code is going to break if `isCurrent()` doesn't exist.

https://github.com/jason-fox/iotagent-node-lib/tree/feature/isCurrent

**Comment by fgalan** [comment]
Usage of isCurrentNgsi():

https://github.com/telefonicaid/iotagent-json/search?q=isCurrentNgsi

**Comment by fgalan** [comment]
> The `isCurrentNgsi()` method hasn't been removed, but it will always be `true`, so the NGSI-v1 specific stuff in the JSON IoT Agent will be uncovered lava code. It is up to you which way round you want to break the code base:
> 
>   * iot-agent-node-lib doesn't support NGSI-v1 but the IoT Agent does.
>   * IoT Agent doesn't support NGSI-v1 but iot-agent-node-lib  does.
> 
> Obviously such a situation is only temporary.

As third alternative, we can use this PR branch (test-ngsiv1-removal) to remove NGSI-v1 IOTA code, re-scoping it's purpose (from just testing to actual code modification). Thus, it would be merged at the end (rolling back the modifications in packages.json before, as last step).

File: lib/commonBindings.js [hunk 1]
```diff
@@ -92,10 +92,7 @@ function guessType(attribute, device) {
     }
 
     if (attribute === constants.TIMESTAMP_ATTRIBUTE) {
-        if (iotAgentLib.configModule.isCurrentNgsi()) {
-            return constants.TIMESTAMP_TYPE_NGSI2;
-        }
-        return constants.TIMESTAMP_TYPE;
```
Comment (fgalan) [inline lib/commonBindings.js:98]: If `TIMESTAMP_TYPE` constant is not used in any place, we could remove it from the source file.

File: lib/iotaUtils.js [hunk 2]
```diff
@@ -82,17 +82,7 @@ function manageConfiguration(apiKey, deviceId, device, objMessage, sendFunction,
     }
 
     function extractAttributes(results, callback) {
```
Comment (fgalan) [inline lib/iotaUtils.js:84]: After simplifying this function, it has ended basically in a pass-through to the callback function.

Maybe we can simplify even more, removeing extractAttributes() function?

File: lib/commonBindings.js [hunk 3]
```diff
@@ -92,10 +92,7 @@ function guessType(attribute, device) {
     }
```
Comment (fgalan) [inline lib/commonBindings.js:92]: CHANGES_NEXT_RELEASE entry is needed. Something like this:

```
- Remove: NGSI-v1 specific behaviours
```

Comment (mapedraza) [reply lib/commonBindings.js:92]: Fixed in https://github.com/telefonicaid/iotagent-json/pull/533/commits/fd29166d7c8e9df6b9a8ffad02b335ef63d1246f

File: lib/iotaUtils.js (hunk 2 above)
Comment (mapedraza) [reply lib/iotaUtils.js]: Fixed in https://github.com/telefonicaid/iotagent-json/pull/533/commits/fd29166d7c8e9df6b9a8ffad02b335ef63d1246f

File: lib/commonBindings.js (hunk 1 above)
Comment (mapedraza) [reply lib/commonBindings.js:98]: Fixed in https://github.com/telefonicaid/iotagent-json/pull/533/commits/c904dfbc857e2fbcef584e279998f7d52e838d92

**Review by fgalan — COMMENTED** [review]
LGTM to code changes, although not ready to be merged yet (first packages.json should be changed back).

I understand same or similar changes has to be mirrored in PR https://github.com/telefonicaid/iotagent-ul/pull/467

**Comment by fgalan** [comment]
> LGTM to code changes, although not ready to be merged yet (first packages.json should be changed back).

Change done in 286588d

**Review by mapedraza — APPROVED** [review]
LGTM

**Pull request merged**